feat: use debug logging when in debug mode (#128)

xremming opened the pull request:
Not 100% sure how to test changes easily. Decided to also include a few small cleanups of the code in the same PR.

After this PR, re-running a failed job with debugging will cause `mise-action` to always output logging at debug level which in turn should help debugging issues that happened in CI (motivated by my Friday debugging of [jdx/mise#2725](https://github.com/jdx/mise/issues/2725)).

Fixes: #128

xremming commented:
@jdx Anything needed from me to get this merged?

jdx commented on src/index.ts:
```diff
@@ -89,7 +94,7 @@ async function restoreMiseCache(): Promise<void> {
   core.info(`mise cache restored from key: ${cacheKey}`)
 }
 
-async function setupMise(version: string | undefined): Promise<void> {
```
isn't this usually set to `undefined`?

xremming replied:
I thought so too originally but the `core.getInput` actually returns an empty string when the value for the given variable is not set. This passes the type checks.

jdx commented:
I ran it with debug enabled and it works: https://github.com/jdx/mise-action/actions/runs/11326474236?pr=129

jdx commented on src/index.ts:
```diff
@@ -40,8 +40,13 @@ async function setEnvVars(): Promise<void> {
+    ? 'debug'
+    : core.getInput('log_level') || 'info'
+
+  set('MISE_LOG_LEVEL', logLevel)
```
if the user didn't specify debug mode or input log_level we should not set `MISE_LOG_LEVEL` at all

xremming replied:
The old (current) code defaults to info when debug level is not set so I kept that behaviour. I agree that maybe it makes more sense to not set the `MISE_LOG_LEVEL` when the user has not set `log_level`.

I'll make that change.

jdx commented:
so I think at least for now we should only set the log level inside of our action, not set it for future actions. Otherwise I think it would add too much noise when doing unrelated things that just happen to call mise shims for example.

jdx commented:
> Hmm. Okay I think I understand what you mean. This PR doesn't change the current behaviour which is to export the log level to all further steps too.

that is fine in the scenario where a user explicitly sets log_level. It's not ok for us to set it automatically just because debugging was enabled on the action generally. I think it's fine within our step but it would create a lot of noise that is likely not what the user would want in unrelated steps.

jdx commented on src/index.ts:
```diff
@@ -151,10 +149,17 @@ const mise = async (args: string[]): Promise<number> =>
       core.getInput('install_dir') ||
       process.cwd()
+    const env = core.isDebug()
+      ? { ...process.env, MISE_LOG_LEVEL: 'debug' }
```
```suggestion
      ? { MISE_LOG_LEVEL: 'debug', ...process.env }
```

if the log level is already set (say, to "trace") we should respect that

xremming replied:
Amended the fix to the commit.

xremming commented:
Looks like you just fixed the cache_save bug that I had fixed in the branch so I rebased and dropped the commit (and squashed some of my changes). Let me know if you want me to rebase again and clean the commits.
